skywalking/trace/span.py

```diff
@@ -117,6 +117,8 @@ def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
+        if isinstance(exc_type, BaseException):
+            self.raised()
```

Comment:
Will this stop the propagation of the exception? We'd rather not to swallow the exceptions from the original codes, that's why we need to raise them again after we captured them, like this

```python
            try:
                 res = _open(this, fullurl, data, timeout)
                 span.tag(Tag(key=tags.HttpMethod, val=fullurl.get_method()))
                 span.tag(Tag(key=tags.HttpUrl, val=fullurl.full_url))
                 span.tag(Tag(key=tags.HttpStatus, val=res.code))
                 if res.code >= 400:
                     span.error_occurred = True
             except HTTPError as e:
                 span.raised()
                 raise e  # <- this one
```

Reply:
No this will not stop propagation.

Reply:
> No this will not stop propagation.

Thanks for clarifying, LGTM